server/src/main/java/io/deephaven/server/appmode/ApplicationServiceGrpcImpl.java

```diff
@@ -209,7 +210,7 @@ private synchronized void propagateUpdates() {
         updatedFields.clear();
         final FieldsChangeUpdate update = builder.build();
 
-        subscriptions.forEach(sub -> sub.send(update));
+        subscriptions.removeIf(subscription -> !subscription.send(update));
```

Comment:
Do we need any of the side-effects that onClose previously provided? (We aren't calling onClose now.)

* `session.removeOnCloseCallback(subscription)`
* `subscription.notifyObserverAborted()`

Reply:
No you're right, though the broken behavior won't be visible externally

 * the leaked callback will just leak until the session cleans it up, which isnt ideal
 * the observer notification doesnt matter, since the observer is already by definition failed

I really dont care for this close-is-remove-but-not-cancel pattern we have used here and elsewhere. I am also not sure that what we seem to have here makes sense at all (why does a close() call necessarily mean sending the aborted message, but doesnt do the cleanup from the session?), but I think that cleanup is outside the scope for this set of changes.